frontend/pages/SoftwarePage/SoftwareVersions/SoftwareVersions.tsx

```diff
@@ -1,11 +1,274 @@
 
-// interface ISoftwareVersionsProps {}
+interface IRowProps extends Row {
+  original: {
+    id?: number;
+  };
+}
+
+interface ISoftwareVersionsProps {
+  router: InjectedRouter;
+  isSoftwareEnabled: boolean;
+  query: string;
+  perPage: number;
+  orderDirection: "asc" | "desc";
+  orderKey: string;
+  showVulnerableSoftware: boolean;
+  currentPage: number;
+  teamId?: number;
+}
+
+const SoftwareVersions = ({
+  router,
+  isSoftwareEnabled,
+  query,
+  perPage,
+  orderDirection,
+  orderKey,
+  showVulnerableSoftware,
+  currentPage,
+  teamId,
+}: ISoftwareVersionsProps) => {
+  const { isSandboxMode, noSandboxHosts, isPremiumTier } = useContext(
+    AppContext
+  );
+
+  // request to get software versions data
+  const {
+    data: softwareVersionsData,
+    isLoading: isSoftwareVersionsLoading,
+    isError: isSoftwareVersionsError,
+  } = useQuery<
+    ISoftwareVersionsResponse,
+    Error,
+    ISoftwareVersionsResponse,
+    ISoftwareQueryKey[]
+  >(
+    [
+      {
+        scope: "software",
+        page: currentPage,
+        perPage,
+        query,
+        orderDirection,
+        orderKey,
+        teamId,
+        vulnerable: showVulnerableSoftware,
+      },
+    ],
+    ({ queryKey }) => softwareAPI.getSoftwareVersions(queryKey[0]),
+    {
+      keepPreviousData: true,
+      // stale time can be adjusted if fresher data is desired based on
+      // software inventory interval
+      staleTime: 30000,
+    }
+  );
+
+  // determines if a user be able to search in the table
+  const searchable =
+    isSoftwareEnabled &&
+    (!!softwareVersionsData?.software ||
+      query !== "" ||
+      showVulnerableSoftware);
+
+  const softwareTableHeaders = useMemo(
+    () =>
+      generateSoftwareVersionsTableHeaders(
+        router,
+        isPremiumTier,
+        isSandboxMode,
+        teamId
+      ),
+    [isPremiumTier, isSandboxMode, router, teamId]
+  );
+
+  const handleVulnFilterDropdownChange = (isFilterVulnerable: string) => {
+    router.replace(
+      getNextLocationPath({
+        pathPrefix: PATHS.SOFTWARE_TITLES,
+        routeTemplate: "",
+        queryParams: {
+          query,
+          teamId,
+          orderDirection,
+          orderKey,
+          vulnerable: isFilterVulnerable,
+          page: 0, // resets page index
+        },
+      })
+    );
+  };
+
+  const handleRowSelect = (row: IRowProps) => {
+    // const hostsBySoftwareParams = {
+    //   software_id: row.original.id,
+    //   team_id: teamId,
+    // };
+
+    // const path = hostsBySoftwareParams
+    //   ? `${PATHS.MANAGE_HOSTS}?${buildQueryStringFromParams(
+    //       hostsBySoftwareParams
+    //     )}`
+    //   : PATHS.MANAGE_HOSTS;
+
+    // router.push(path);
+    // TODO: navigation to software details page.
+    console.log("selectedRow", row.id);
+  };
+
+  const onQueryChange = () => {
+    console.log("onQueryChange");
+  };
+
+  const renderSoftwareCount = useCallback(() => {
```

Comment:
This seems like something we could potentially simplify now that we are not relying on a separate API call for the count. But I'm in favor of not reinventing the wheel so I think it is a good idea to stick with the prior pattern for now.